### PR TITLE
Thrifit protocol to support multi cast stream sink

### PR DESCRIPTION
**Pull request by dirtysalt**
ref: #1675

### CODE REVIEW

File: gensrc/thrift/DataSinks.thrift [hunk 1]
```diff
@@ -73,6 +84,11 @@ struct TDataStreamSink {
   3: optional bool ignore_not_found
 }
 
+struct TMCastDataStreamSink {
```
Comment (Seaven) [inline gensrc/thrift/DataSinks.thrift:87]:
```suggestion
struct TMulitCastDataStreamSink {
```

**Review by Seaven — APPROVED** [review]

**Comment by dirtysalt** [comment]
run starrocks_be_unittest

**Comment by dirtysalt** [comment]
run starrocks_clang-format

**Comment by dirtysalt** [comment]
run starrocks_fe_unittest

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**